src/server/http/http_test.go

```diff
@@ -40,7 +40,7 @@ func TestRouting(t *testing.T) {
 	for _, test := range testData {
 		t.Run(test.name, func(t *testing.T) {
 			ctx := pctx.TestContext(t)
-			s := New(0, func(ctx context.Context) *client.APIClient { return nil })
+			s := New(ctx, 0, func(ctx context.Context) *client.APIClient { return nil })
```

Comment:
These tests fail now because New can't handle a factory that returns a nil client anymore.  We will have to move the tests to NewRealEnv to get a real pach client, or do `if clientFactory(ctx) == nil { skipping rest gateway setup }`.  It would be nice to test the REST gateway here though.

Reply:
I don't see a way around a real grpc connection for grpc-gateway registration to work

Reply:
I agree with that.  Since the test doesn't cover the gateway, registration could be skipped.  Or you could realenv.NewRealEnv() to get a real client here.  (RealEnv doesn't require k8s, but does have a real GRPC client.)  Another option, if the Registration handlers accept a grpc.ClientConnInterface, would be to return a stub that implements that from a nil pach client.  (Implement Invoke and NewStream, just have them return status.Error(codes.Unimplemented, "unimplemented") or something.)

Reply:
Moved routing test one level up to src/server - did have to make Server.http.Server field public for test to access

Reply:
Ah, if it's because of an import cycle you can just change the package declaration to `package http_test` instead of `package http`.

Reply:
For just http_test.go? I'd still have to make the field public